### PR TITLE
pipeline version 1

### LINKED ISSUE #4: Train/val/test/evaluation pipeline + prepare baselines
Kind of a boring task, but we'll need to assemble the other 3 parts, as well as prepare some baseline results. In order for our new architecture to be successful, we'd probably have to demonstrate at least one of the following:
a) better accuracy/whatever eval metric was used in baseline
b) faster convergence to what was used in the baseline.

### PR DESCRIPTION
**Pull request by zyy2929**

### CODE REVIEW

File: main_two.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+###################################
+############ Train ############
+###################################
+print('###################################')
+print('############## Train ##############')
+print('###################################')
+
+
+# training process
+model = BasicCNN
```
Comment (abhon) [inline main_two.py]: There are some parameters you would need to put in here for BasicCNN to work, check my code for the parameters!

File: main_two.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+train_loader =
+valid_loader =
+test_loader =
+
+###################################
+######### optim & criter ##########
+###################################
+
+optimizer = optim.Adam(model.parameters())
```
Comment (abhon) [inline main_two.py]: I don't think we need to set this here, same with epoch count, etc. You wrote a function to take all those parameters so we can decide those later when we call the function.(Suppose we want SGD instead of Adam for instance) Same with the Criterions(97-104) below

**Review by abhon — CHANGES_REQUESTED** [review]
I'm a little confused about the relationship between the two files you have here, are there two training/validation functions? Could you explain how you want the pipeline to work with these two files? I guess my main question is what is the differences between the two files.

File: main_two.py (hunk 1 above)
Comment (zyy2929) [reply main_two.py]: The main_two.py uses train and validation function from train_valid.py  (from train_valid import train, validation).  Train and validation function in train_valid.py return training loss, accuracy... every training epoch and the main_two.py collects those returned values and store them. Eg, history['train_loss'] for 3 epochs would be like [1.00, 0.99, 0.98], each element in history['train_loss'] indicates the training loss value every epoch.

Comment (zyy2929) [reply main_two.py]: This file is not complete. The parameters of BasicCNN may depend on task 3 feature engineering and the dataset and will be changed later.

File: main_two.py (hunk 2 above)
Comment (zyy2929) [reply main_two.py]: Sure!
